src/app/campaign/campaign.component.ts

```diff
@@ -69,9 +78,17 @@ export class CampaignComponent implements OnInit, OnDestroy {
         const id = +params.get('id');
         this.store.dispatch(new campaignActions.CampaignLoad({ id }));
       } else {
-        this.store.dispatch(new campaignActions.CampaignNew());
```

Comment:
Both duplicated and new campaigns are routed to `/campaign/new`. Angular 7.2 supports a `routerLink` state we can use to pass data between routes: https://netbasal.com/set-state-object-when-navigating-in-angular-7-2-b87c5b977bb.